src/sync/remote_mongo_collection.cpp

```diff
@@ -407,24 +261,230 @@ void MongoCollection::find_one_and_replace(const bson::BsonDocument& filter_bson
 void MongoCollection::find_one_and_delete(const bson::BsonDocument& filter_bson,
                                           MongoCollection::FindOneAndModifyOptions options,
                                           std::function<void(util::Optional<bson::BsonDocument>, util::Optional<AppError>)> completion_block)
+{
+    find_one_and_delete_bson(filter_bson, options, handle_document_response(completion_block));
+}
+
+void MongoCollection::find_one_and_delete(const bson::BsonDocument& filter_bson,
+                                          std::function<void(util::Optional<bson::BsonDocument>, util::Optional<AppError>)> completion_block)
+{
+    find_one_and_delete(filter_bson, {}, completion_block);
+}
+
+void MongoCollection::find_bson(const bson::BsonDocument& filter_bson,
+                                FindOptions options,
+                                std::function<void(util::Optional<AppError>, util::Optional<bson::Bson>)> completion_block)
+{
+    try {
+        auto base_args = m_base_operation_args;
+        base_args["query"] = filter_bson;
+
+        if (options.limit) {
+            base_args["limit"] = *options.limit;
+        }
+
+        if (options.projection_bson) {
+            base_args["project"] = *options.projection_bson;
+        }
+
+        if (options.sort_bson) {
+            base_args["sort"] = *options.sort_bson;
+        }
+
+        m_service->call_function("find",
```

Comment:
Hmm, is this rebased on my patch that binds the MongoCollection to a SyncUser? Check out how this looks in the current v10 branch https://github.com/realm/realm-object-store/blob/6b44209e6fcac0137e193c96444f93c50d184d06/src/sync/remote_mongo_collection.cpp#L115

Reply:
Good catch! Must have botched something when copy-pasting stuff.